components/script/dom/mutationobserver.rs

```diff
@@ -135,6 +136,26 @@ impl MutationObserver {
                         }
                         // Step 3.1.2
                         let paired_string = if registered.options.attribute_old_value {
+                            old_value.clone()
+                        } else {
+                            None
+                        };
+                        // Step 3.1.1
+                        let idx = interestedObservers.iter().position(|&(ref o, _)|
+                            &**o as *const _ == &*registered.observer as *const _);
+                        if let Some(idx) = idx {
+                            interestedObservers[idx].1 = paired_string;
+                        } else {
+                            interestedObservers.push((DomRoot::from_ref(&*registered.observer),
+                                                      paired_string));
```

Comment:
The code duplication here is somewhat unfortunate, can we move the common code somewhere, maybe a local closure?

That being said, it's not much code... Your call I guess.